src/lib/models/containers.ts

```
@@ -0,0 +1,533 @@
+
+		// Update trace mappings only if this container isn't already tracked
+		if (!this.allContainers.has(container.id)) {
+			this.allContainers.set(container.id, container);
+
+			const traceId = container.getTraceId();
+			if (traceId) {
+				// Track which containers belong to which trace
+				if (!this.traceIdToContainerIds.has(traceId)) {
+					this.traceIdToContainerIds.set(traceId, new Set());
+				}
+				this.traceIdToContainerIds.get(traceId)!.add(container.id);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)